src/execution_plan/ops/op_update.c

```diff
@@ -178,6 +229,83 @@ static OpResult UpdateInit(OpBase *opBase) {
 	return OP_OK;
 }
 
+static void _EvalEntityUpdates(EntityUpdateCtx *ctx, GraphContext *gc,
+							   Record r) {
+	Schema *s = NULL;
+	const char *label = NULL;
+	bool update_index = false;
+	int label_id = GRAPH_NO_LABEL;
+
+	// Get the type of the entity to update. If the expected entity was not
+	// found, make no updates but do not error.
+	RecordEntryType t = Record_GetType(r, ctx->record_idx);
+	if(t == REC_TYPE_UNKNOWN) return;
+
+	// Make sure we're updating either a node or an edge.
+	if(t != REC_TYPE_NODE && t != REC_TYPE_EDGE) {
+		QueryCtx_SetError("Update error: alias '%s' did not resolve to a graph entity",
+						  ctx->alias);
+		QueryCtx_RaiseRuntimeException();
+	}
+
+	GraphEntity *entity = Record_GetGraphEntity(r, ctx->record_idx);
+	GraphEntityType type = (t == REC_TYPE_NODE) ? GETYPE_NODE : GETYPE_EDGE;
+
+	// If the entity is a node, set its label if possible.
+	if(type == GETYPE_NODE) {
+		Node *n = (Node *)entity;
+		label = n->label;  // n->label will be set if specified in query string.
+
+		if(label == NULL) {
+			label_id = Graph_GetNodeLabel(gc->g, ENTITY_GET_ID(entity));
+			s = GraphContext_GetSchemaByID(gc, label_id, SCHEMA_NODE);
+			if(s) label = Schema_GetName(s);
+		}
+
+		n->label = label;
+		n->labelID = label_id;
+	}
+
+	uint exp_count = array_len(ctx->exps);
+	for(uint i = 0; i < exp_count; i++) {
+		EntityUpdateEvalCtx *update_ctx = ctx->exps + i;
+		SIValue new_value = AR_EXP_Evaluate(update_ctx->exp, r);
+
+		/* Determine whether we must update the index for this set of updates.
+		 * If at least one property being updated is indexed, each node will be reindexed. */
+		if(!update_index && label) {
+			Attribute_ID attr_id = update_ctx->attribute_id;
+			const char *field = GraphContext_GetAttributeString(gc, attr_id);
+			update_index = GraphContext_GetIndex(gc, label, field, IDX_ANY);
+			if(update_index && i > 0) {
```

Comment:
```suggestion
			if(update_index && (i > 0)) {
```
Just for readability